### PR TITLE
core/ts-version_issue_473_add_system_error_to_erros

### PR DESCRIPTION
**Pull request by GreatGaryCooper**
## Check list  

- [ ] All Merge Request pipelines should be passed.
- [ ] Make sure all the changes are necessary and correct.
- [ ] Make sure no needed change missing (all necessary changes are made).
- [ ] Merge Request should contains only changes for one purpose (one feature or bug or refactoring).
- [ ] All necessary unit tests are written.
- [ ] Critical changes should be tested locally (or on staging).

### CODE REVIEW

**Comment by WilliamHawkingStephen** [comment]
nice job for separating errors format :+1: 
 we dont need to pass errors in invalidParsingErrors ?

**Comment by WilliamHawkingStephen** [comment]
can you explain for me whats the purpose of isOptionalError ?

File: ts-version/src/domain/entities/order/policies/cancel.ts [hunk 1]
```diff
@@ -13,11 +17,14 @@ export class CancelOrderPolicy extends Policy<Order> {
 
   isNewStatusValid = (newOrder: Order) => {
     if (newOrder.status !== OrderStatus.cancelled)
-      throw new Error("New status is not cancelled");
+      throw new CancelOrderPolicyNewOrderStatusNotValid({
```
Comment (WilliamHawkingStephen) [inline ts-version/src/domain/entities/order/policies/cancel.ts:20]: - suggestion: can we have a better name definition for this cancel order policy? i think it is a little bit confusing

File: ts-version/src/domain/errors/system-errors/cancel-order-policy.error.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { OrderStatus } from "@type/order.type";
+import { BaseSystemError } from "./base-system-error";
+
+export class CancelOrderPolicyNewOrderStatusNotValid extends BaseSystemError {
```
Comment (WilliamHawkingStephen) [inline ts-version/src/domain/errors/system-errors/cancel-order-policy.error.ts:4]: why we have two classes for handling cancel order error policy ? 
can we have a class that has a SRP for cancel orders policy?

**Review by WilliamHawkingStephen — APPROVED** [review]

File: ts-version/src/domain/errors/system-errors/cancel-order-policy.error.ts (hunk 2 above)
Comment (GreatGaryCooper) [reply ts-version/src/domain/errors/system-errors/cancel-order-policy.error.ts:4]: Will apply your suggestion we talked about in another cart, thanks.

File: ts-version/src/domain/entities/order/policies/cancel.ts (hunk 1 above)
Comment (GreatGaryCooper) [reply ts-version/src/domain/entities/order/policies/cancel.ts:20]: Yes you're right, will apply those suggestions in another cart.

**Review by franken5tein — APPROVED** [review]

**Pull request merged**